lib/dynamics_crm/xml/attributes.rb

```diff
@@ -111,6 +113,13 @@ def build_xml(key, value, type)
       def render_value_xml(type, value)
         xml = ""
         case type
+        when "ArrayOfEntity"
+          raise "We can only serialize Entities inside of ArrayOfEntity" unless value.all?{|a| a.is_a?(DynamicsCRM::XML::Entity)}
+          xml << %Q{
+          <c:value i:type="a:ArrayOfEntity">
+              #{value.map(&->(_) { _.to_xml({in_array: true}) }).join}
```

Comment:
Heh, yeah that's just some sneaky one liner to run self (`&`) pass it into a lambda (`->{}`) and running to_xml on self, which is set to `_`, maybe a little silly looking, but ruby is fun. 💃